pkgs/applications/editors/emacs/build-support/generic.nix

```diff
@@ -51,7 +55,7 @@ stdenv.mkDerivation (finalAttrs: ({
 
   buildInputs = [emacs texinfo] ++ packageRequires ++ buildInputs;
   propagatedBuildInputs = packageRequires;
```

Comment:
For elisp package builders (genericBuild, trivialBuild, elpaBuild, melpaBuild), all attributes can only be in one of two states:
- they are overridden by user-specified ones
- or they are merged with user-specified ones.

I think presumably some attributes should be classified into the second class.  So I change those attributes in this PR and leave others untouched.

I do not think there is a principled way to classify them though.  Any thoughts from people who are familiar with builders in other ecosystems?